shelley/chain-and-ledger/executable-spec/test/Rules/TestUtxow.hs

```diff
@@ -1,176 +1,153 @@
-                                       , signal = tx
-                                       , target = UTxOState u' _ _ _}) =
-          created u' tx pp stp == consumed u tx pp stk
-        created u tx pp stp =
-            balance u
-          + _txfee (_body tx)
-          + (deposits pp stp (toList $ _certs $ _body tx))
-        consumed u tx pp stk =
-            balance u
-          + keyRefunds pp stk (_body tx)
+preserveBalance
+  :: PParams
+  -> [(StakeCreds, StakePools, SourceSignalTarget UTXOW)]
+  -> Property
+preserveBalance pp tr =
+  conjoin $
+    map createdIsConsumed tr
+  where
+    createdIsConsumed (stk, stp, SourceSignalTarget
+                                   { source = UTxOState u _ _ _
+                                   , signal = tx
+                                   , target = UTxOState u' _ _ _}) =
+      created u' stp tx == consumed u stk tx
+
+    created u stp_ tx =
+        balance u
+      + _txfee (_body tx)
+      + deposits pp stp_ (toList $ _certs $ _body tx)
+
+    consumed u stk_ tx =
+        balance u
+      + keyRefunds pp stk_ (_body tx)
 
 -- | Preserve balance restricted to TxIns and TxOuts of the Tx
-preserveBalanceRestricted :: Property
-preserveBalanceRestricted = withTests (fromIntegral numberOfTests) . property $ do
-  t <- forAll (trace @UTXOW testGlobals traceLen `ofLengthAtLeast` 1)
-  let
-    tr = sourceSignalTargets t
-    UtxoEnv _ pp stk stp _ = _traceEnv t
-
-  assertAll (createdIsConsumed pp stk stp) tr
-
-  where createdIsConsumed pp stk stp (SourceSignalTarget
-                                       { source = UTxOState u _ _ _
-                                       , signal = tx
-                                       , target = UTxOState _ _ _ _}) =
-          inps u tx == outs (_body tx) pp stk stp
-        inps u tx = balance $ (_inputs $ _body tx) <| u
-        outs tx pp stk stp =
-            balance (txouts tx)
-          + _txfee tx
-          + depositChange pp stk stp (toList $ _certs tx) tx
-
-        depositChange pp stk stp certs txb =
-            deposits pp stp certs
-          - (refunded pp stk txb + decayed pp stk txb)
-        refunded pp stk txb = keyRefunds pp stk txb
-        decayed pp stk txb  = runShelleyBase $ decayedTx pp stk txb
+preserveBalanceRestricted
+  :: PParams
+  -> [(StakeCreds, StakePools, SourceSignalTarget UTXOW)]
+  -> Property
+preserveBalanceRestricted pp tr =
+  conjoin $
+    map createdIsConsumed tr
+  where
+    createdIsConsumed (stk, stp, SourceSignalTarget
+                                   { source = UTxOState u _ _ _
+                                   , signal = tx
+                                   , target = UTxOState _ _ _ _}) =
+      inps u tx == outs stk stp (_body tx)
+
+    inps u tx = balance $ (_inputs $ _body tx) <| u
+
+    outs stk_ stp_ tx =
+        balance (txouts tx)
+      + _txfee tx
+      + depositChange stk_ stp_ (toList $ _certs tx) tx
+
+    depositChange stk_ stp_ certs txb =
```

Comment:
@mgudemann -> here I removed the `decayed` part since that is factored into `refunded` already:

`deposits pp stp certs - (refunded pp stk txb + decayed pp stk txb)`